pkg/agent/agent.go

```diff
@@ -345,6 +349,14 @@ func (i *Initializer) setupGatewayInterface() error {
 		return err
 	}
 
+	if i.networkConfig.TrafficEncapMode.IsNetworkPolicyOnly() {
+		// In policy-only mode, Node IP is also assigned to local gateway for masquerade.
+		i.nodeConfig.GatewayConfig = &config.GatewayConfig{Name: i.hostGateway, MAC: link.Attrs().HardwareAddr, IP: i.nodeConfig.NodeIPAddr.IP}
```

Comment:
@jianjuns @suwang48404 @wenyingd If masquerading here is to send packets back to OVS pipeline for network policy enforcement, this is not going to work as the packets has no original source ip and of_port, network policy flows cannot identify packets need to filter.
Did I miss anything?

Reply:
Hi Quan, u r correct. Network Policy does not work for external service in policy-only mode. I plan to file an issue once this is checked in, explaining the problem and potential remedy.

Reply:
I believe nodeIP assigning to gw0 is discussed and resolved during our offline chatting?

Resolving.